.github/workflows/deploy.yml

```diff
@@ -8,44 +8,23 @@ jobs:
     name: Building distribution
     strategy:
       matrix:
-        include:
-          - os: ubuntu-latest
-            python: '3.8'
-            numpy: '1.16'
-            source: false
-          - os: ubuntu-latest
-            python: '3.9'
-            numpy: '1.19'
-            source: false
-          - os: ubuntu-latest
-            python: '3.10'
-            numpy: '1.21'
-            source: true
-          - os: macos-latest
-            python: '3.8'
-            numpy: '1.16'
-            source: false
-          - os: macos-latest
-            python: '3.9'
-            numpy: '1.19'
-            source: false
-          - os: macos-latest
-            python: '3.10'
-            numpy: '1.21'
-            source: false
-          - os: windows-latest
-            python: '3.8'
-            numpy: '1.16'
-            source: false
-          - os: windows-latest
-            python: '3.9'
-            numpy: '1.19'
-            source: false
-          - os: windows-latest
-            python: '3.10'
-            numpy: '1.21'
-            source: false
+        source: false
+        os: ['ubuntu-latest', 'windows-latest', 'macos-latest']
+        versions: 
+          - {python: '3.8', numpy: '1.19'}
+          - {python: '3.8', numpy: '1.23'}
+          - {python: '3.8', numpy: 'none'}
 
+          - {python: '3.9', numpy: '1.19'}
+          - {python: '3.9', numpy: '1.23'}
+          - {python: '3.9', numpy: 'none'}
+
+          - {python: '3.10', numpy: '1.21'}
+          - {python: '3.10', numpy: '1.23'}
+          - {python: '3.10', numpy: 'none'}
+
+          - {python: '3.11', numpy: '1.23'}
+          - {python: '3.11', numpy: 'none'}
```

Comment:
Thanks for the changes. In this setup `source` would always be `false`, right? If this is the case, the source distribution would not be built.

Second, I do not quite understand the purpose for multiple NumPy versions for each Python version. To my understanding the lowest NumPy version should be compatible with all minor versions above, correct? So the lowest possible version should be sufficient here.

Apart from that, you may also add yourself to `CONTRIB.rst`, if you like.

Reply:
hey @padix-key sorry I just was in the middle of editing and i got pulled away and forgot to finish this...  this isn't done yet. I'll address your concerns later today.

Reply:
OK, then I will review the changes, when you are ready. You do not need to hurry.